developer/js/source/lexical-model-compiler/build-trie.ts

```diff
@@ -138,7 +138,7 @@ function _parseWordList(wordlist: WordList, source:  WordListSource): void {
     }
     wordsSeenInThisFile.add(wordform);
 
-    wordlist[wordform] = (wordlist[wordform] || 0) + count;
+    wordlist[wordform] = (isNaN(wordlist[wordform]) ? 0 : wordlist[wordform] || 0) + count;
```

Comment:
Does that fix the problem or merely apply a hammer to the symptom?

Looking at the definition of `Object` I think all other current problematic properties are going to be camelCased or __snakyCamelCased__ so are unlikely to cause trouble.

We may need to be a little aware of other libraries extending the `Object` prototype although perhaps not such an issue in the compiler. (This could be unit-tested pretty easily.)

Reply:
I'm not so sure I'd call it a "hammer".

> We may need to be a little aware of other libraries extending the Object prototype although perhaps not such an issue in the compiler. (This could be unit-tested pretty easily.)

That's exactly why I wrote it this way.  They'd have to explicitly extend the Object prototype with a pre-initialized property of type `number` for it to cause any issue.  If they added a function... well, `isNaN(function) == false`, so we'd interpret that (rightly) as a starting value of 0 for our associative array.  I'd want the same thing for `string`, `boolean`, etc to be extra safe, but as it turns out...

![Screen Shot 2021-02-23 at 8 20 40 AM](https://user-images.githubusercontent.com/25213402/108790767-16137580-75b0-11eb-845f-82229cef0285.png)

Looks like I'd need to do even more.  Perhaps I should also add a `typeof` condition-check to be extra sure.  (But `typeof NaN == "number"`, so it _would_ need to be an _extra_ check.)

Our only use of that `Object` is for its use as an "associative array" / map / dictionary structure.  Here, for this specific case, we'd _want_ to obliterate any default values for the Object type if and when they existed.

I mean, I _guess_ we could explicitly filter for just `"constructor"`, but this honestly seems like the safer path forward.